### ISSUE TITLE
Build fails

### ISSUE THREAD

**Issue by shirhatti** [ligershark/Kulture#4]
Build fails when no file is selected OR when selected file is nested deep in the folder structure.

Build only works when selected file is in the same directory as project.json

**Comment by sayedihashimi** [comment]
Using the design time host will help for these situations.

**Comment by shirhatti** [comment]
Fixed issue on Mac 6236d5361d0010a79c41ace99ce6e111157855df

Investigating windows

**Comment by schotime** [comment]
Also, for windows in the ASP.NET.sublime-build file you have `k.cmd build`. I believe this should be `kpm .cmd build`. Same in the .py file.

**Comment by jchannon** [comment]
Using this on a Windows box (the Visual Studio 14 CTP2 VM) and running Build from ST it errors with `Unable to load application or execute command 'build'` I think @schotime is correct as it builds fine in the cmd line using `kpm build`

**Comment by shirhatti** [comment]
Has already been fixed in 377067d08625c9840fa92be058b51a5e13aaff4f
Moving to release soon

**Comment by jchannon** [comment]
:thumbsup:

**Comment by shirhatti** [comment]
Just pushed the changes to release ec0b64f7a3ea8434f916bae17289b05e7271e9d0

**Comment by jchannon** [comment]
Updated the package in Sublime but it still looks like its trying to run `k build` not `kpm build` via `Ctrl+B` however the Run K command Build option works

**Comment by shirhatti** [comment]
Oops. I only checked in the fix for mac. Fixed in eb049e160ae741f823d20a405cfcd7b4c2a6e876

**Comment by jchannon** [comment]
Thats the same commit as before so I dont think it is fixed in Windows?

**Comment by shirhatti** [comment]
I pasted the wrong SHA. I edited my comment above eb049e160ae741f823d20a405cfcd7b4c2a6e876

**Comment by jchannon** [comment]
Cool, let us know when its pushed to sublime repo and i'll retest.

Great work by the way! :balloon:

**Comment by shirhatti** [comment]
Live in Sublime
https://sublime.wbond.net/packages/Kulture

**Comment by schotime** [comment]
You still have to push build when at the level of the project.json file otherwise it complains that you don't have a project.json

**Comment by jchannon** [comment]
Seems to work on Windows for me

**Comment by schotime** [comment]
![image](https://cloud.githubusercontent.com/assets/202923/3792107/35afb616-1b66-11e4-93c0-2f3c19ff0b21.png)

**Comment by ashutoshraina** [comment]
Running K Web from sublime on Windows gives the following 

```
An unhandled exception occurred while processing the request.

MissingMethodException: Method not found: 'Microsoft.Framework.Runtime.ILibraryExport Microsoft.Framework.Runtime.ILibraryManager.GetAllExports(System.String)'.
Microsoft.AspNet.Mvc.Razor.Compilation.RoslynCompilationService.GetApplicationReferences()

Stack Query Cookies Headers Environment
MissingMethodException: Method not found: 'Microsoft.Framework.Runtime.ILibraryExport Microsoft.Framework.Runtime.ILibraryManager.GetAllExports(System.String)'.
Microsoft.AspNet.Mvc.Razor.Compilation.RoslynCompilationService.GetApplicationReferences()
Microsoft.AspNet.Mvc.Razor.Compilation.RoslynCompilationService.Compile(String content)
Microsoft.AspNet.Mvc.Razor.RazorCompilationService.CompileCore(IFileInfo file)
Microsoft.AspNet.Mvc.Razor.RazorCompilationService.<>c__DisplayClass0.<Compile>b__2()
Microsoft.AspNet.Mvc.Razor.CompilerCache.GetOrAdd(IFileInfo file, Func`1 compile)
Microsoft.AspNet.Mvc.Razor.RazorCompilationService.Compile(IFileInfo file)
Microsoft.AspNet.Mvc.Razor.VirtualPathRazorPageFactory.CreateInstance(String path)
Microsoft.AspNet.Mvc.Razor.RazorViewEngine.CreateViewEngineResult(ActionContext context, String viewName, Boolean partial)
Microsoft.AspNet.Mvc.Razor.RazorViewEngine.FindView(ActionContext context, String viewName)
Microsoft.AspNet.Mvc.Rendering.CompositeViewEngine.FindView(ActionContext context, String viewName, Boolean partial)
Microsoft.AspNet.Mvc.Rendering.CompositeViewEngine.FindView(ActionContext context, String viewName)
Microsoft.AspNet.Mvc.ViewResult.FindView(IViewEngine viewEngine, ActionContext context, String viewName)
Microsoft.AspNet.Mvc.ViewResult.<ExecuteResultAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.AspNet.Mvc.ReflectedActionInvoker.<InvokeActionResult>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.AspNet.Mvc.ReflectedActionInvoker.<InvokeActionResultFilter>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
Microsoft.AspNet.Mvc.ReflectedActionInvoker.<InvokeActionResultWithFilters>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.AspNet.Mvc.ReflectedActionInvoker.<InvokeActionAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
Microsoft.AspNet.Mvc.MvcRouteHandler.<RouteAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.AspNet.Routing.Template.TemplateRoute.<RouteAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.AspNet.Routing.RouteCollection.<RouteAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.AspNet.Builder.RouterMiddleware.<Invoke>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.AspNet.RequestContainer.ContainerMiddleware.<Invoke>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.AspNet.Diagnostics.ErrorPageMiddleware.<Invoke>d__1.MoveNext()
```

**Comment by shirhatti** [comment]
@ashutoshraina Looks like the version of the runtime you are using is well out-of-date. There have been some API changes since then.

Try running a `kvm upgrade`. That should fix your error

**Comment by ashutoshraina** [comment]
Yup, that worked.  The read me had listed a specific version, may be the kvm upgrade instruction should be there as well. :+1:

**Comment by shirhatti** [comment]
@schotime @ashutoshraina I have pushed out a temporary fix in cdf77070d5e1221f9451f161c5ab06925576c189

![capture6](https://cloud.githubusercontent.com/assets/4734691/3801494/cbb25aa8-1c04-11e4-9b86-842a22d891c8.PNG)

Still investigating the best way to solve this

**Comment by sayedihashimi** [comment]
I made some changes that should fix this. More info in comments at #22. Could you please verify on your platform? I tried on Mac + Windows and it looks to be working good.

I will close this item. Let's continue further discussion at #22.
